app_dart/lib/src/request_handlers/github_webhook.dart

```diff
@@ -440,21 +441,51 @@ class GithubWebhook extends RequestHandler<Body> {
     }
   }
 
-  Future<void> _checkBaseRef(
+  /// Validate the base and head refs of the PR.
+  Future<void> _validateRefs(
     GitHub gitHubClient,
     PullRequest pr,
   ) async {
-    if (pr.base.ref != 'master') {
-      final String body = await _getWrongBaseComment(pr.base.ref);
-      final RepositorySlug slug = pr.base.repo.slug();
+    final RepositorySlug slug = pr.base.repo.slug();
+    String body;
+    const List<String> releaseChannels = <String>[
+      'stable',
+      'beta',
+      'dev',
+    ];
+    // Close PRs that use a release branch as a source.
+    if (releaseChannels.contains(pr.head.ref)) {
+      body = config.wrongHeadBranchPullRequestMessage(pr.head.ref);
       if (!await _alreadyCommented(gitHubClient, pr, slug, body)) {
         await gitHubClient.pullRequests.edit(
           slug,
           pr.number,
-          base: 'master',
+          state: 'closed',
         );
         await gitHubClient.issues.createComment(slug, pr.number, body);
       }
+      return;
+    }
+    if (pr.base.ref == kDefaultBranchName) {
+      return;
+    }
+    final RegExp candidateTest = RegExp(r'flutter-\d+\.\d+-candidate\.\d+');
+    if (pr.base.ref == pr.head.ref &&
+        candidateTest.hasMatch(pr.base.ref) &&
+        candidateTest.hasMatch(pr.head.ref)) {
+      // This is most likely a release branch
+      return;
+    }
```

Comment:
This will leave PRs alone for anyone who opens a patch against a release branch. 

Can we at least have this check that the author is a member of the Flutter org? I know it's possible to get that with the GraphQL API, it's probably possible with the rest API as well.  The graphQL query for it is here: https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handlers/check_for_waiting_pull_requests_queries.dart#L40

Reply:
My only concern is that the reason this was opened in the first place is that people opened seemingly helpful pull requests against the wrong branch, and they would sometimes get merged because a reviewer didn't notice which branch it was for.

Reply:
I can if you feel strongly about it, but I doubt someone would accidentally open a PR from a release candidate branch on their fork to an upstream release candidate branch of the exact same name. I can't think of an easier way to do this than:

1. `git checkout flutter-1.20-candidate.7`
2. maybe make commit some local changes
3. `git push -u origin flutter-1.20-candidate.7`
4. In GitHub, open a PR.
5. Change the base branch to be `flutter-1.20-candidate.7`.

Reply:
However, I guess we should still comment on the issue, even if we don't rebase.

Reply:
I'm playing with this endpoint, I can see the flutter org at https://api.github.com/users/dnfield/orgs, but I get an empty array when I GET https://api.github.com/users/christopherfujino/orgs

Reply:
Anyone who wants to see their desired commit hotfixed into a release branch could do that, especially if they're reading our public documentation about how to do it :) 

I don't think it would happen often, but that's part of the issue. If we just comment that's probably good enough.

Reply:
If that's their intention, the bot specifically says to rebase it back to what it was previously if you're doing a hotfix, and the user would assume this is part of the procedure and change the base branch (I think the hotfix docs even have that instruction). The bot isn't really enforcing policy, but warning of a possible (probably?) error. To that end, the bot should certainly still comment (and comments are cheap :) ).

Reply:
added a comment

Reply:
Comment is fine, don't worry about org if it's a pain here.

Reply:
I'll look into it another time. It seems easy to implement, but I'm not sure why my account isn't showing up as a member of Flutter.